src/EFCore.PG/Storage/Internal/Mapping/NpgsqlCharacterTypeMapping.cs

```diff
@@ -64,5 +64,23 @@ protected override void ConfigureParameter(DbParameter parameter)
 
             base.ConfigureParameter(parameter);
         }
+
+        // Needed because we can't use Spans in expression trees
+        static bool EqualsWithoutTrailingWhitespace(string a, string b)
+        {
+            var (i, j) = (0, 0);
+
+            for (i = a.Length - 1; a[i] >= 0 && char.IsWhiteSpace(a[i]); i--) ;
+            for (j = b.Length - 1; b[j] >= 0 && char.IsWhiteSpace(b[j]); j--) ;
```

Comment:
Doesn't this throw on a whitespace-only string? 

Should `a[i] >= 0` be `i >= 0`?

Reply:
Oops, yeah, it should... will fix.